src/lucky_router/path_part.cr

```diff
@@ -0,0 +1,46 @@
+# A PathPart represents a single section of a path
+#
+# It can be a static path
+#
+# ```crystal
+# path_part = PathPart.new("users")
+# path_part.path_variable? => false
+# path_part.optional? => false
+# path_part.name => "users"
+# ```
+#
+# It can be a path variable
+#
+# ```crystal
+# path_part = PathPart.new(":id")
+# path_part.path_variable? => true
+# path_part.optional? => false
+# path_part.name => "id"
+# ```
+#
+# It can be optional
+#
+# ```crystal
+# path_part = PathPart.new("?users")
+# path_part.path_variable? => false
+# path_part.optional? => true
+# path_part.name => "users"
+# ```
+struct LuckyRouter::PathPart
+  def self.split_path(path : String) : Array(PathPart)
+    parts = path.split('/')
+    parts.pop if parts.last.blank?
+    parts.map { |part| new(part) }
+  end
+
+  getter part : String
+  getter name : String
+  getter? optional : Bool
+  getter? path_variable : Bool
+
+  def initialize(@part)
+    @name = part.lchop('?').lchop(':')
+    @path_variable = part.starts_with?(':') || part.starts_with?("?:")
+    @optional = part.starts_with?('?')
```

Comment:
What do you think about making these methods instead? I think this will also speed things up a lot because right now it will do these lchops and `starts_with?` check for every part, even when it isn't needed.

But if you have methds like `optional?` and `path_variable?` it might be clearer and much faster.